packages/builder/src/utils/wagmi.ts

```diff
@@ -6,135 +6,72 @@ import {
 import { publicProvider } from "wagmi/providers/public";
-import PublicGoodsNetworkIcon from "common/src/icons/PublicGoodsNetwork.svg";
+import {
+  pgn,
+  pgnTestnet,
+  devChain1,
+  devChain2,
+  avalanche,
+  avalancheFuji,
+  fantom,
+  fantomTestnet,
+} from "common/src/chains";
 import { polygon, polygonMumbai } from "@wagmi/core/chains";
-import { FantomFTMLogo } from "../assets";
 
 // RPC keys
 const alchemyId = process.env.REACT_APP_ALCHEMY_ID!;
 const infuraId = process.env.REACT_APP_INFURA_ID!;
 
-export const avalanche: Chain = {
-  ...avalancheChain,
-  rpcUrls: {
-    default: {
-      http: [
-        "https://avalanche-mainnet.infura.io/v3/1e0a90928efe4bb78bb1eeceb8aacc27",
-      ],
-    },
-    public: {
-      http: ["https://api.avax.network/ext/bc/C/rpc"],
-    },
-  },
-};
-
-export const avalancheFuji: Chain = {
-  ...avalancheFujiChain,
-  rpcUrls: {
-    default: {
-      http: [
-        "https://avalanche-fuji.infura.io/v3/1e0a90928efe4bb78bb1eeceb8aacc27",
-      ],
-    },
-    public: {
-      http: ["https://api.avax-test.network/ext/bc/C/rpc"],
-    },
-  },
-};
-
-export const fantom: Chain = {
-  ...fantomChain,
-  rpcUrls: {
-    default: {
-      http: ["https://rpcapi.fantom.network/"],
-    },
-    public: {
-      http: ["https://rpcapi.fantom.network/"],
-    },
-  },
-  iconUrl: FantomFTMLogo,
-};
-
-export const fantomTestnet: Chain = {
-  ...fantomTestnetChain,
-  rpcUrls: {
-    default: {
-      http: ["https://rpc.testnet.fantom.network/"],
-    },
-    public: {
-      http: ["https://rpc.testnet.fantom.network/"],
-    },
-  },
-  iconUrl: FantomFTMLogo,
-};
-
-export const pgn: Chain = {
-  id: 424,
-  name: "PGN",
-  network: "pgn",
-  iconUrl: PublicGoodsNetworkIcon,
-  nativeCurrency: {
-    decimals: 18,
-    name: "Ether",
-    symbol: "ETH",
-  },
-  rpcUrls: {
-    default: { http: ["https://rpc.publicgoods.network"] },
-    public: { http: ["https://rpc.publicgoods.network"] },
-  },
-  blockExplorers: {
-    default: {
-      name: "pgnscan",
-      url: "https://explorer.publicgoods.network",
-    },
-  },
+const availableChains: { [key: string]: Chain } = {
+  dev1: devChain1,
+  dev2: devChain2,
+  mainnet,
+  fantom,
+  optimism,
+  pgn,
+  arbitrum,
+  avalanche,
+  polygon,
+  goerli,
+  fantomTestnet,
+  pgnTestnet,
+  arbitrumGoerli,
+  polygonMumbai,
+  avalancheFuji,
 };
 
-const chainsAvailable: Chain[] = [];
-export const pgnTestnet: Chain = {
-  id: 58008,
-  name: "PGN Testnet",
-  network: "pgn testnet",
-  iconUrl: PublicGoodsNetworkIcon,
-  nativeCurrency: {
-    decimals: 18,
-    name: "Ether",
-    symbol: "ETH",
-  },
-  rpcUrls: {
-    default: { http: ["https://sepolia.publicgoods.network"] },
-    public: { http: ["https://sepolia.publicgoods.network"] },
-  },
-  blockExplorers: {
-    default: {
-      name: "pgnscan",
-      url: "https://explorer.sepolia.publicgoods.network",
-    },
-  },
-  testnet: true,
-};
+const enabledChains: Chain[] = [];
```

Comment:
Could all this be moved into  function so as to minimize the amount of calls done as side effect of module loading, and make it test-friendly?